### PR TITLE
PHP - Drive v2 snippets

### PR DESCRIPTION
**Pull request by RajeshGogo**
# Description
drive_create_folder, drive_create_shortcut, drive_download_file, drive_export_pdf, drive_fetch_app_data_folder, drive_fetch_changes, drive_fetch_start_page_token, drive_list_app_data, drive_move_file_to_folder, drive_search_files, drive_share_file, drive_touch_file, drive_upload_app_data, drive_upload_basic, drive_upload_to_folder, drive_upload_with_conversion, 

Fixed(234633738)

## Has it been tested?
- [X] Development testing done
- [X] Unit or integration test implemented

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have performed a peer-reviewed with team member(s)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: drive/snippets/drive-v2/src/createShortcut.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START drive_create_shortcut]
+require_once 'vendor/autoload.php';
+putenv('GOOGLE_APPLICATION_CREDENTIALS=/opt/lampp/htdocs/google-webspace/workspace-348506-241f41f76ce5.json');
```
Comment (sqrrrl) [inline drive/snippets/drive-v2/src/createShortcut.php:20]: Leftover from local testing?

**Review by sqrrrl — COMMENTED** [review]
See comments on #99 -- same apply throughout these files (use explicit imports, don't overwrite local vars, etc)

File: drive/snippets/drive-v2/src/createFolder.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+ // [START drive_create_folder]
+require_once 'vendor/autoload.php';
+putenv('GOOGLE_APPLICATION_CREDENTIALS=/opt/lampp/htdocs/google-webspace/workspace-348506-241f41f76ce5.json');
+
+function createFolder() {
+    $client = new Google\Client();
+    $client->useApplicationDefaultCredentials();
+    $client->addScope(Google\Service\Drive::DRIVE);
+    $driveService = new Google_Service_Drive($client);
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/createFolder.php:26]: Please use the namespaced versions of the classes. The underscored names are there only to preserve backwards compatibility.

```suggestion
    $driveService = new Google\Service\Drive($client);
```

File: drive/snippets/drive-v2/src/createFolder.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+   
+    $fileMetadata = new Google_Service_Drive_DriveFile(array(
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/createFolder.php:28]:
```suggestion
    $fileMetadata = new Google\Service\Drive\DriveFile(array(
```

File: drive/snippets/drive-v2/src/createFolder.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        'name' => 'A',
+        'mimeType' => 'application/vnd.google-apps.folder'));
+    $file = $driveService->files->create($fileMetadata, array(
+        'fields' => 'id'));
+    printf("Folder ID: %s\n", $file->id);
+}
+ // [END drive_create_folder]
+createFolder();
+?>
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/createFolder.php:37]: It's idiomatic to leave off the ending `?>` in PHP unless you're rendering HTML
```suggestion

```

File: drive/snippets/drive-v2/src/createShortcut.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+function createShortcut()
+{
+        $client = new Google\Client();
+        $client->useApplicationDefaultCredentials();
+        $client->addScope(Google\Service\Drive::DRIVE);
+        $driveService = new Google_Service_Drive($client);
+        
+        $fileMetadata = new Google_Service_Drive_DriveFile(array(
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/createShortcut.php:29]:
```suggestion
        $driveService = new Google\Service\Drive($client);
        
        $fileMetadata = new Google\Service\Drive\DriveFile(array(
```

File: drive/snippets/drive-v2/src/createShortcut.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            'name' => 'Project plan',
+            'mimeType' => 'application/vnd.google-apps.drive-sdk'));
+        $file = $driveService->files->create($fileMetadata, array(
+            'fields' => 'id'));
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/createShortcut.php:33]: Short-array syntax is more common in modern PHP
```suggestion
        $file = $driveService->files->create($fileMetadata, [
            'fields' => 'id'
        ]);
```

File: drive/snippets/drive-v2/src/downloadFile.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    // [START drive_download_file]
+require_once 'vendor/autoload.php';
+putenv('GOOGLE_APPLICATION_CREDENTIALS=/opt/lampp/htdocs/google-webspace/workspace-348506-241f41f76ce5.json');
+
+function downloadFile() {
+
+    $client = new Google\Client();
+    $client->useApplicationDefaultCredentials();
+    $client->addScope(Google\Service\Drive::DRIVE);
+    $driveService = new Google_Service_Drive($client);
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/downloadFile.php:26]:
```suggestion
    $driveService = new Google\Service\Drive($client);
```

File: drive/snippets/drive-v2/src/downloadFile.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+    $fileId = '0BwwA4oUTeiV1UVNwOHItT0xfa2M';
+    // [END_EXCLUDE]
+    $response = $driveService->files->get($fileId, array(
+        'alt' => 'media'));
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/downloadFile.php:31]:
```suggestion
    $response = $driveService->files->get($fileId, [
        'alt' => 'media'
    ]);
```

File: drive/snippets/drive-v2/src/exportPdf.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+// [START drive_export_pdf]          
+require_once 'vendor/autoload.php';
+putenv('GOOGLE_APPLICATION_CREDENTIALS=/opt/lampp/htdocs/google-webspace/workspace-348506-241f41f76ce5.json');
+     // TODO - Currently broken due to PHP client confiscating 'mimeType' param
+function exportPdf($realFileId)
+{
+         $client = new Google\Client();
+         $client->useApplicationDefaultCredentials();
+         $client->addScope(Google\Service\Drive::DRIVE);
+         $driveService = new Google_Service_Drive($client);
```
Comment (bshaffer) [inline drive/snippets/drive-v2/src/exportPdf.php:27]:
```suggestion
         $driveService = new Google\Service\Drive($client);
```

**Comment by sqrrrl** [comment]
Please switch to explicit use statements + see Brent's comments. Also need to resolve merge conflicts

**Pull request closed without merging**